libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/factory/RoadObjectFactory.kt

```diff
@@ -37,6 +37,27 @@ object RoadObjectFactory {
             }
     }
 
+    fun List<UpcomingRoadObject>.getUpdatedObjectsAhead(
```

Comment:
it would be nice to add unit tests for this, but it's getting late, will add it tomorrow if I have enough time. Or add in a follow up PR if I don't